### ISSUE TITLE
error in addGeneScoreMat 

### ISSUE THREAD

**Issue by parkjams** [GreenleafLab/ArchR#317]
Hello - I am attempting to use the ArchR package to analyze several scATAC-seq data sets (generated from 10X chromium platform). From what I've seen and have heard, this package seems quite powerful and user-friendly. 

Unfortunately, when I try to create an arrow file, I get an error when it gets to the section of determining .addGeneScoreMat. This leads to an a blank ArrowFile being created. 

I've attached the log file for this instance. At first I tried running the function "createArrowFiles()" with the parameters "thread" = 16. In another thread for another error, I noticed that running "thread" = 1 may help. Unfortunately that did not do the trick. 

I CAN create an ArrowFile if I choose to NOT run the AddGeneScoreMat. However, I would like to integrate these scATACseq data sets with corresponding scRNAseq data sets, and as I understand it, it is necessary to have gene scores.

Please advise on how to move forward with this, i.e., determine gene scores. Any input is much appreciated! 

Thank you for developing this package! I look forward to being able to fully explore the functionalities of this package!

log info
***********************************
[createArrows .txt](https://github.com/GreenleafLab/ArchR/files/5249209/createArrows.txt)

***********************************
sessionInfo()
R version 3.6.2 (2019-12-12)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.3 LTS

Matrix products: default
BLAS:   /home/jpark/local/R-3.6.2/lib/libRblas.so
LAPACK: /home/jpark/.local/share/r-miniconda/envs/r-reticulate/lib/libmkl_rt.so

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] nabor_0.5.0                       gridExtra_2.3                     Rsamtools_2.2.3                  
 [4] BSgenome.Hsapiens.UCSC.hg38_1.4.1 BSgenome_1.54.0                   rtracklayer_1.46.0               
 [7] Biostrings_2.54.0                 XVector_0.26.0                    ArchR_0.9.5                      
[10] magrittr_1.5                      rhdf5_2.30.1                      Matrix_1.2-18                    
[13] data.table_1.13.0                 devtools_2.3.1                    usethis_1.6.1                    
[16] Hmisc_4.4-1                       Formula_1.2-3                     survival_3.2-3                   
[19] lattice_0.20-41                   destiny_3.0.1                     dplyr_1.0.2                      
[22] sctransform_0.2.1                 pheatmap_1.0.12                   scater_1.14.6                    
[25] ggplot2_3.3.2                     scran_1.14.6                      RCurl_1.98-1.2                   
[28] GSVA_1.34.0                       RColorBrewer_1.1-2                limma_3.42.2                     
[31] genefilter_1.68.0                 GSVAdata_1.22.0                   hgu95a.db_3.2.3                  
[34] org.Hs.eg.db_3.10.0               GSEABase_1.48.0                   graph_1.64.0                     
[37] annotate_1.64.0                   XML_3.99-0.3                      AnnotationDbi_1.48.0             
[40] Seurat_3.2.1                      NMF_0.23.0                        bigmemory_4.5.36                 
[43] cluster_2.1.0                     rngtools_1.5                      pkgmaker_0.31.1                  
[46] registry_0.5-1                    superheat_1.0.0                   biomaRt_2.42.1                   
[49] monocle3_0.2.1                    SingleCellExperiment_1.8.0        SummarizedExperiment_1.16.1      
[52] DelayedArray_0.12.3               BiocParallel_1.20.1               matrixStats_0.56.0               
[55] GenomicRanges_1.38.0              GenomeInfoDb_1.22.1               IRanges_2.20.2                   
[58] S4Vectors_0.24.4                  Biobase_2.46.0                    BiocGenerics_0.32.0              

loaded via a namespace (and not attached):
  [1] rsvd_1.0.3               vcd_1.4-7                ica_1.0-2                class_7.3-17             ps_1.3.4                
  [6] foreach_1.5.0            lmtest_0.9-38            rprojroot_1.3-2          crayon_1.3.4             laeken_0.5.1            
 [11] MASS_7.3-53              nlme_3.1-149             backports_1.1.10         rlang_0.4.7              ROCR_1.0-11             
 [16] readxl_1.3.1             irlba_2.3.3              callr_3.4.4              smoother_1.1             bit64_4.0.5             
 [21] glue_1.4.2               processx_3.4.4           vipor_0.4.5              haven_2.3.1              tidyselect_1.1.0        
 [26] rio_0.5.16               fitdistrplus_1.1-1       tidyr_1.1.2              zoo_1.8-8                GenomicAlignments_1.22.1
 [31] xtable_1.8-4             RcppHNSW_0.3.0           bibtex_0.4.2.2           cli_2.0.2                zlibbioc_1.32.0         
 [36] rstudioapi_0.11          miniUI_0.1.1.1           sp_1.4-2                 rpart_4.1-15             RcppEigen_0.3.3.7.0     
 [41] shiny_1.5.0              BiocSingular_1.2.2       xfun_0.17                askpass_1.1              pkgbuild_1.1.0          
 [46] pcaMethods_1.78.0        tibble_3.0.3             ggrepel_0.8.2            listenv_0.8.0            png_0.1-7               
 [51] future_1.18.0            withr_2.2.0              bitops_1.0-6             ranger_0.12.1            plyr_1.8.6              
 [56] cellranger_1.1.0         e1071_1.7-3              dqrng_0.2.1              pillar_1.4.6             fs_1.5.0                
 [61] scatterplot3d_0.3-41     TTR_0.24.2               DelayedMatrixStats_1.8.0 xts_0.12.1               vctrs_0.3.4             
 [66] ellipsis_0.3.1           generics_0.0.2           tools_3.6.2              foreign_0.8-76           beeswarm_0.2.3          
 [71] munsell_0.5.0            proxy_0.4-24             fastmap_1.0.1            compiler_3.6.2           pkgload_1.1.0           
 [76] abind_1.4-5              httpuv_1.5.4             sessioninfo_1.1.1        plotly_4.9.2.1           GenomeInfoDbData_1.2.2  
 [81] edgeR_3.28.1             deldir_0.1-29            later_1.1.0.1            BiocFileCache_1.10.2     jsonlite_1.7.1          
 [86] ggplot.multistats_1.0.0  scales_1.1.1             pbapply_1.4-3            carData_3.0-4            lazyeval_0.2.2          
 [91] promises_1.1.1           spatstat_1.64-1          car_3.0-9                doParallel_1.0.15        latticeExtra_0.6-29     
 [96] goftest_1.2-2            spatstat.utils_1.17-0    reticulate_1.16          checkmate_2.0.0          openxlsx_4.1.5          
[101] cowplot_1.1.0            statmod_1.4.34           Rtsne_0.15               forcats_0.5.0            uwot_0.1.8              
[106] igraph_1.2.5             htmltools_0.5.0          memoise_1.1.0            locfit_1.5-9.4           viridisLite_0.3.0       
[111] digest_0.6.25            assertthat_0.2.1         mime_0.9                 rappdirs_0.3.1           bigmemory.sri_0.1.3     
[116] RSQLite_2.2.0            future.apply_1.6.0       remotes_2.2.0            blob_1.2.1               shinythemes_1.1.2       
[121] splines_3.6.2            labeling_0.3             Rhdf5lib_1.8.0           Cairo_1.5-12.2           hms_0.5.3               
[126] colorspace_1.4-1         base64enc_0.1-3          ggbeeswarm_0.6.0         nnet_7.3-14              Rcpp_1.0.5              
[131] RANN_2.6.1               fansi_0.4.1              VIM_6.0.0                R6_2.4.1                 ggridges_0.5.2          
[136] lifecycle_0.2.0          zip_2.1.1                curl_4.3                 leiden_0.3.3             testthat_2.3.2          
[141] robustbase_0.93-6        desc_1.2.0               RcppAnnoy_0.0.16         iterators_1.0.12         stringr_1.4.0           
[146] htmlwidgets_1.5.1        polyclip_1.10-0          purrr_0.3.4              mgcv_1.8-33              globals_0.12.5          
[151] openssl_1.4.3            htmlTable_2.1.0          patchwork_1.0.1          codetools_0.2-16         prettyunits_1.1.1       
[156] dbplyr_1.4.4             gridBase_0.4-7           RSpectra_0.16-0          gtable_0.3.0             DBI_1.1.0               
[161] tensor_1.5               httr_1.4.2               KernSmooth_2.23-17       stringi_1.5.3            progress_1.2.2          
[166] reshape2_1.4.4           farver_2.0.3             viridis_0.5.1            ggthemes_4.2.0           hexbin_1.28.1           
[171] boot_1.3-25              BiocNeighbors_1.4.2      geneplotter_1.64.0       DEoptimR_1.0-8           bit_4.0.4               
[176] jpeg_0.1-8.1             spatstat.data_1.4-3      pkgconfig_2.0.3          knitr_1.29

**Comment by rcorces** [comment]
Can you recapitulate this error with the tutorial dataset?

**Comment by parkjams** [comment]
Thanks for the response, Ryan. 

To answer your question, I was NOT able to reproduce the error with the tutorial dataset. Prior to trying reproduce the error with the tutorial dataset, I tried a few things including:
1) reinstalling the "GenomicRanges" package, as the error associated with the addGeneScoreMatrix() seemed to be related to the nearest.distance(), a function in the GenomicRanges package.
2) restarting my RStudio session. 

Upon the restart, I was able to create an ArrowFile WITH the GeneScoreMatrix for both the tutorial data set and my own datasets. So far things are running smoothly. I don't know if it was the restart, the reinstall or both that fixed the issue, but thigns are ok for now. 

Thanks again for the follow up!
